README.md

```diff
@@ -160,41 +160,49 @@ Suppose we need to process intensity/mask file p1_y2_r68_c1.ome.tif :
 ```
 
 ## Building from source
-To build Nyxus from source, make sure you clone the Github repository with the `--recurse-submodules` option to clone all the necessary dependencies.
 
-```
-git clone --recurse-submodules https://github.com/PolusAI/nyxus.git
-```
+Nyxus can either be build inside a `conda` environment or independently outside of it. For the later case, we provide a script to make it easier to download and build all the necessary dependencies.
 
-Nyxus relies on [libTIFF](http://www.libtiff.org) as an external dependency which is readily available on most Unix-based OSs via the system package manager. 
-
-For Debian-based distros, such as Ubuntu, this can be installed via:
+### Inside Conda
+To build Nyxus from source, make sure you clone the Github repository with the `--recurse-submodules` option to clone FastLoader and Hedgehog, which are two necessary dependencies. Nyxus uses a CMake build system. Below is an example of how to build Nyxus inside a `conda` environment.
 
 ```bash
-sudo apt install libtiff-dev
+git clone --recurse-submodules https://github.com/PolusAI/nyxus.git
+cd Nyxus
+mkdir build
+cd build
+conda install -y --file -c conda-forge ci-utils/envs/conda_cpp.txt --file ci-utils/envs/conda_gpu.txt --file ci-utils/envs/conda_py.txt
+cmake -DBUILD_CLI=ON -DUSEGPU=ON -DCMAKE_PREFIX_PATH=$CONDA_PREFIX -DCMAKE_INSTALL_PREFIX=$CONDA_PREFIX ..
+make -j4
 ```
 
-For CentOS:
+Note that `-DBUILD_CLI=ON` tells Nyxus to build the command line interface and `-DUSEGPU=ON` enables CUDA support if a CUDA supported device is avialable on the host. 
```

Comment:
I think I found the source of the header file issues. The problem is that in `cu-utils/envs/conda_cpp.txt`, `compilers=1.1.1` pins clang to version 10, which has incompatible header files. Removing 1.1.1 allows the build to proceed (with clang 13) on my MacOS 12.5.1 machine. Note that (as per the stackoverflow post I linked above) with clang 10 I can't even compile hello world. Also note that /usr/bin/clang is version 13

FYI I've always thought it was a good idea to separate out pinned dependencies into a *_pinned.txt file, and then explain why they are pinned. Pinned versions (especially for very fundamental dependencies such as compilers) can occasionally force the solver to choose bizarre plans, so unless you really need 1.1.1 I'd say un-pin it.

Reply:
Thanks for taking a deep-dive. The original reason to pin the compiler version was to restrict GCC < 10, since CUDA does not support GCC 10 and beyond. I guess since we decided not to support GPU builds on Mac (reasons beyond our control), I should split the compiler dependency to separate files based on OS. I will test it out once in my local machines and update the PR.

Reply:
Thanks for splitting it out into a separate file. Also, just something to think about, but if you really only need gcc (i.e. you don't need clang) then it might be good to just install gcc directly instead of the compilers package.